lib/config_files/base_model.rb

```diff
@@ -26,11 +27,41 @@ def save(changes_only: false)
 
     def load
       self.data = @parser.parse(@file_handler.read(@file_path))
+      @loaded = true
+    end
+
+    # powerfull method that sets any value in grub config. It try to be
+    # smart to at first modify existing value, then replace commented out code
+    # and if even that doesn't work, then append it at the end
+    # @note prefer to use specialized methods of children
+    def generic_set(key, value)
+      modify(key, value) || uncomment(key, value) || add_new(key, value)
```

Comment:
I tend to agree that replacing the commented out code is a good approach, but if I think it twice I'm not sure if it's 100% correct in all situations. Usually, commented values are used to indicate the default value and replacing it instead of adding a line below can be considered as some kind of "destroying documentation".